prog/main.go

```diff
@@ -138,6 +142,41 @@ type appFlags struct {
 	consulInf       string
 }
 
+type containerLabelFiltersFlag struct {
+	apiTopologyOptions []app.APITopologyOption
+	exclude            bool
+}
+
+func (c *containerLabelFiltersFlag) String() string {
+	return fmt.Sprint(c.apiTopologyOptions)
+}
+
+func (c *containerLabelFiltersFlag) Set(flagValue string) error {
+	newAPITopologyOption, err := c.toAPITopologyOption(flagValue)
+	c.apiTopologyOptions = append(c.apiTopologyOptions, newAPITopologyOption)
+	return err
+}
+
+func (c *containerLabelFiltersFlag) toAPITopologyOption(flagValue string) (app.APITopologyOption, error) {
+	var indices []int
+	if indices = colonFinder.FindStringIndex(flagValue); len(indices) == 0 {
+		return app.APITopologyOption{}, fmt.Errorf("No unescaped colon found. This is needed to separate the title from the label")
+	} else if len(colonFinder.FindAllStringIndex(flagValue, -1)) > 1 {
```

Comment:
Hold on a second. I already reviewed this. Why not calling `colonFinder.FindAllIndex` and checking that he number of indices is 1? Why do you need to call  `FindStringIndex` and `FindAllStringIndex` separately?